auth/oidc.go

```diff
@@ -692,6 +692,8 @@ func (op *OIDCProvider) startDiscoverySync(ctx context.Context, discoveryURL str
 		return err
 	}
 	go func() {
+		// use DbConsoleLogConfig as nil in 3.1 branch only. This is stored on database in >= 3.2
+		ctx := base.NewNonCancelCtxForDatabase(ctx, nil).Ctx
```

Comment:
I don't think we need to make this function diverge across versions. There's no reason why this signature can't be the same, only taking in `ctx`, and the actual implementation sets `ctx.DbConsoleLogConfig` in 3.1

This is nessesary for log streaming using per-database log levels/log keys, and passing `nil` in here potentially breaks that for logging coming from this goroutine.